src/main/antlr4/org/hisp/dhis/parser/expression/antlr/Expression.g4

```diff
@@ -498,6 +504,15 @@ EMPTY
 WS  :   [ \t\n\r]+
     ;
 
+// Comments
+
+SINGLE_LINE_COMMENT
+    :   ( '<--' ~[\r\n]* '-->') -> skip;
```

Comment:
Suggest no space after the (, for consistency with the syntax in the rest of the file.

Reply:
resolved

Reply:
Please chose another syntax. This unnecessarily collides with syntax we already use for something else. Even if parser can handle it, it can be confusing to users. Also 6 marker characters for a single line comment misses the point of having another syntax to a multi line comment. The entire reason languages have these two is that a multi-line needs an end marker and that usually the markers are 2 or more characters to not require escaping within but still be able to use almost any character sequence. The single line then has the role of being concise by only requiring a start marker as the end it marked by the line break. Also many languages chose a single character start marker like `%` again to have the advantage of being concise compared to the multi-line comment syntax. A longer single line with start and end marker isn't worth having. In that case you rather can have only the multi-line syntax. A single line should also not have an end marker because then it can be used wrongly by using a line break before the end marker. This sort of possible mistake when commenting should be avoided by choosing a syntax that makes it impossible. Hence, this is why single line ends implicitly on the line break or end of input.

Reply:
only keeping multi-line comments. Single line `'//' `may also conflict with other operators. Moreover
the whole point for this is to provide dhis2 implementers **(most of the time they are not developers)** a way to explain the logic of expression so I think only multi-line would serve the purpose.